From 3a5d5c079be5c3adcfdb03a2d82d7c7c68c0300d Mon Sep 17 00:00:00 2001
From: Amit Prakash Ambasta <amit.prakash.ambasta@gmail.com>
Date: Fri, 11 Mar 2022 16:09:16 +0530
Subject: [PATCH 23/26] [PATCH 30/30] bgo#816975 Fix build on x86

Signed-off-by: Amit Prakash Ambasta <amit.prakash.ambasta@gmail.com>
---
 modules/fdlibm/src/math_private.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/modules/fdlibm/src/math_private.h b/modules/fdlibm/src/math_private.h
index 51d79f9c2e..fafd7d6fc1 100644
--- a/modules/fdlibm/src/math_private.h
+++ b/modules/fdlibm/src/math_private.h
@@ -30,7 +30,11 @@
  * Adapted from https://github.com/freebsd/freebsd-src/search?q=__double_t
  */
 
+#if defined __FLT_EVAL_METHOD__ && (__FLT_EVAL_METHOD__ == 2)
+typedef long double      __double_t;
+#else
 typedef double      __double_t;
+#endif
 typedef __double_t  double_t;
 
 /*
-- 
2.35.1

