diff --git a/libappstream-glib/as-tag.c b/libappstream-glib/as-tag.c
index 9e84381..a1b8490 100644
--- a/libappstream-glib/as-tag.c
+++ b/libappstream-glib/as-tag.c
@@ -37,8 +37,11 @@
 
 #ifdef HAVE_GPERF
   /* we need to define this now as gperf just writes a big header file */
-  const struct tag_data *_as_tag_from_gperf (const char *tag, guint etag);
+  #pragma GCC diagnostic push
+  #pragma GCC diagnostic ignored "-Wconversion"
+  #pragma GCC diagnostic ignored "-Wmissing-prototypes"
   #include "as-tag-private.h"
+  #pragma GCC diagnostic pop
 #endif
 
 /**
