diff --git a/CMakeLists.txt b/CMakeLists.txt
index bd60be12..a96c2c74 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,13 @@
 cmake_minimum_required(VERSION 2.8.8 FATAL_ERROR)
 include (GenerateExportHeader)
+if (NOT WIN32)
+    include (GNUInstallDirs)
+else()
+    set(CMAKE_INSTALL_BINDIR "bin")
+    set(CMAKE_INSTALL_LIBDIR "lib")
+    set(CMAKE_INSTALL_INCLUDEDIR "include")
+endif()
+
 if(NOT (CMAKE_MAJOR_VERSION LESS 2))
     if(POLICY CMP0022)
         cmake_policy(SET CMP0022 OLD)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 9be9f922..26d996f7 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -185,9 +185,9 @@ endforeach()
 
 install(TARGETS libcryptominisat5
     EXPORT ${CRYPTOMINISAT5_EXPORT_NAME}
-    LIBRARY DESTINATION "${CMAKE_INSTALL_PREFIX}/lib"
-    ARCHIVE DESTINATION "${CMAKE_INSTALL_PREFIX}/lib"
-    PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_PREFIX}/include/cryptominisat5"
+    LIBRARY DESTINATION "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}"
+    ARCHIVE DESTINATION "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}"
+    PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR}/cryptominisat5"
 )
 
 if (NOT ONLY_SIMPLE)
@@ -246,7 +246,7 @@ if (NOT ONLY_SIMPLE)
     )
     install(TARGETS cryptominisat5
         EXPORT ${CRYPTOMINISAT5_EXPORT_NAME}
-        RUNTIME DESTINATION bin
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
     )
     SET(CPACK_PACKAGE_EXECUTABLES "cryptominisat5")
 
@@ -256,7 +256,7 @@ if (NOT ONLY_SIMPLE)
         )
         install(TARGETS scalmc
             EXPORT ${CRYPTOMINISAT5_EXPORT_NAME}
-            RUNTIME DESTINATION bin
+            RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
         )
         SET(CPACK_PACKAGE_EXECUTABLES "scalmc")
     endif()
@@ -267,7 +267,7 @@ target_link_libraries(cryptominisat5_simple
 )
 install(TARGETS cryptominisat5_simple
     EXPORT ${CRYPTOMINISAT5_EXPORT_NAME}
-    RUNTIME DESTINATION bin
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
 )
 SET(CPACK_PACKAGE_EXECUTABLES "cryptominisat5_simple")
 
