diff -urN ropemode-0.2/ropemode/decorators.py ropemode-0.2.new/ropemode/decorators.py
--- ropemode-0.2/ropemode/decorators.py	2011-03-03 16:07:36.000000000 +0530
+++ ropemode-0.2.new/ropemode/decorators.py	2015-01-06 20:17:46.705635392 +0530
@@ -16,7 +16,7 @@
 
     def _show(self, message):
         if message is None:
-            print message
+            print(message)
         else:
             self.message(message)
 
@@ -27,7 +27,7 @@
     def newfunc(*args, **kwds):
         try:
             func(*args, **kwds)
-        except Exception, e:
+        except Exception as e:
             trace = str(traceback.format_exc())
             short = 'Ignored an exception in ropemode hook: %s' % \
                     _exception_message(e)
@@ -47,11 +47,12 @@
                     exceptions.ModuleSyntaxError,
                     exceptions.BadIdentifierError)
 
+
 def _exception_handler(func):
     def newfunc(*args, **kwds):
         try:
             return func(*args, **kwds)
-        except exceptions.RopeError, e:
+        except exceptions.RopeError as e:
             short = None
             if isinstance(e, input_exceptions):
                 short = _exception_message(e)
@@ -60,9 +61,11 @@
     newfunc.__doc__ = func.__doc__
     return newfunc
 
+
 def _exception_message(e):
     return '%s: %s' % (e.__class__.__name__, str(e))
 
+
 def rope_hook(hook):
     def decorator(func):
         func = lisphook(func)
diff -urN ropemode-0.2/ropemode/filter.py ropemode-0.2.new/ropemode/filter.py
--- ropemode-0.2/ropemode/filter.py	2011-03-03 16:07:36.000000000 +0530
+++ ropemode-0.2.new/ropemode/filter.py	2015-01-06 20:18:45.440825602 +0530
@@ -23,7 +23,7 @@
         except exceptions.ResourceNotFoundError:
             continue
         if resource.is_folder():
-            matches = set(filter(lambda item: resource.contains(item), all))
+            matches = set([item for item in all if resource.contains(item)])
         else:
             matches = set([resource])
         if first == '+':
diff -urN ropemode-0.2/ropemode/interface.py ropemode-0.2.new/ropemode/interface.py
--- ropemode-0.2/ropemode/interface.py	2012-05-17 02:15:24.000000000 +0530
+++ ropemode-0.2.new/ropemode/interface.py	2015-01-06 20:19:35.594987360 +0530
@@ -1,3 +1,4 @@
+import collections
 import os
 
 import rope.base.change
@@ -24,7 +25,7 @@
     def _init_mode(self):
         for attrname in dir(self):
             attr = getattr(self, attrname)
-            if not callable(attr):
+            if not isinstance(attr, collections.Callable):
                 continue
             kind = getattr(attr, 'kind', None)
             if kind == 'local':
@@ -300,7 +301,7 @@
         if modules:
             for i in range(len(modules)):
                 modname = modules[i]
-                if not isinstance(modname, basestring):
+                if not isinstance(modname, str):
                     modname = modname.value()
                 modnames.append(modname)
         else:
