From cc98d5322eac0450266ab6a2aef234471023a25b Mon Sep 17 00:00:00 2001
From: Amit Prakash Ambasta <amit.prakash.ambasta@gmail.com>
Date: Tue, 24 Sep 2019 12:22:33 +0530
Subject: [PATCH] Patches for gcr-non-ui

---
 CMakeLists.txt   | 4 ++--
 core/urlbar.vala | 6 +++---
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8e72f08..6af336c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,12 +53,12 @@ set(PKGS sqlite3 gio-2.0 libsoup-2.4)
 pkg_check_modules(DEPS_GTK REQUIRED
                   gtk+-3.0>=3.12.0
                   webkit2gtk-4.0>=2.16.6
-                  gcr-ui-3>=2.32
+                  gcr-base-3>=2.32
                   libpeas-gtk-1.0
                   json-glib-1.0>=0.12
                   libarchive
                   )
-set(PKGS ${PKGS} gtk+-3.0 libsoup-2.4 gcr-ui-3 libpeas-gtk-1.0 json-glib-1.0 posix)
+set(PKGS ${PKGS} gtk+-3.0 libsoup-2.4 gcr-3 libpeas-gtk-1.0 json-glib-1.0 posix)
 set(EXTRA_VAPIS
     ${CMAKE_SOURCE_DIR}/vapi/config.vapi
     ${CMAKE_SOURCE_DIR}/vapi/libarchive.vapi
diff --git a/core/urlbar.vala b/core/urlbar.vala
index 5f4e866..4bc8a45 100644
--- a/core/urlbar.vala
+++ b/core/urlbar.vala
@@ -46,7 +46,7 @@ namespace Midori {
         Gtk.Button trust;
         [GtkChild]
         Gtk.Label security_status;
-        Gcr.CertificateWidget? details = null;
+        // Gcr.CertificateWidget? details = null;
 
         construct {
             size_allocate.connect (resize);
@@ -301,7 +301,7 @@ namespace Midori {
         void icon_pressed (Gtk.EntryIconPosition position, Gdk.Event event) {
             var tls = ((Browser)get_toplevel ()).tab.tls;
             var certificate = tls != null ? new Gcr.SimpleCertificate (tls.certificate.data) : null;
-            if (details == null) {
+            /* if (details == null) {
                 var icon_area = get_icon_area (position);
                 security.relative_to = this;
                 security.pointing_to = icon_area;
@@ -328,7 +328,7 @@ namespace Midori {
             }
             details.visible = tls != null;
             trust.visible = tls != null && !secure;
-            details.certificate = certificate;
+            details.certificate = certificate; */
             security_status.visible = !secure;
             security.show ();
         }
-- 
2.23.0

