From d42290a9cf6dce08faa148b5eecf9a5799adcc7f Mon Sep 17 00:00:00 2001
From: Amit Prakash Ambasta <amit.prakash.ambasta@gmail.com>
Date: Tue, 15 Dec 2020 14:57:50 +0530
Subject: [PATCH] Removed x11 dependency from meson build

---
 meson.build    | 1 -
 ui/meson.build | 1 -
 2 files changed, 2 deletions(-)

diff --git a/meson.build b/meson.build
index 57b625b..281f5d4 100644
--- a/meson.build
+++ b/meson.build
@@ -55,7 +55,6 @@ endif
 if get_option('gtk')
   gtk_min_version = '3.12'
   gtk_dep = dependency('gtk+-3.0', version: '>=' + gtk_min_version)
-  gtk_x11_dep = dependency('gtk+-x11-3.0', version: '>=' + gtk_min_version)
 endif
 
 # configuration
diff --git a/ui/meson.build b/ui/meson.build
index 0b89b24..741d8b3 100644
--- a/ui/meson.build
+++ b/ui/meson.build
@@ -84,7 +84,6 @@ gcr_ui_deps = [
   gck_dep,
   gcr_base_dep,
   gtk_dep,
-  gtk_x11_dep,
 ]
 
 gcr_ui_cflags = [
-- 
2.29.2

