From 7d1ba8593c2a4f72b52b20035e22d56be4df7a91 Mon Sep 17 00:00:00 2001
From: Amit Prakash Ambasta <amit.prakash.ambasta@gmail.com>
Date: Wed, 27 Mar 2024 17:47:14 +0530
Subject: [PATCH 1/2] bmo#1882942: add missing header for size_t

Signed-off-by: Amit Prakash Ambasta <amit.prakash.ambasta@gmail.com>
---
 layout/style/PseudoStyleType.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/layout/style/PseudoStyleType.h b/layout/style/PseudoStyleType.h
index 6804b500c3..88eb4e4cf8 100644
--- a/layout/style/PseudoStyleType.h
+++ b/layout/style/PseudoStyleType.h
@@ -7,6 +7,7 @@
 #ifndef mozilla_PseudoStyleType_h
 #define mozilla_PseudoStyleType_h
 
+#include <cstddef>
 #include <cstdint>
 #include <iosfwd>
 
-- 
2.44.0

