From f0a3df4e42260d448b0adadc98be8f336f113a35 Mon Sep 17 00:00:00 2001
From: Amit Prakash Ambasta <amit.prakash.ambasta@gmail.com>
Date: Fri, 11 Mar 2022 16:11:32 +0530
Subject: [PATCH 24/26] [PATCH 31/30] bgo#831903 pip don't fail with optional
 deps

Signed-off-by: Amit Prakash Ambasta <amit.prakash.ambasta@gmail.com>
---
 python/mach/mach/site.py | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/python/mach/mach/site.py b/python/mach/mach/site.py
index 941bc17985..187f5b0d96 100644
--- a/python/mach/mach/site.py
+++ b/python/mach/mach/site.py
@@ -918,10 +918,6 @@ class ExternalPythonSite:
 
         if not result.has_all_packages:
             print(result.report())
-            raise Exception(
-                f'The Python packages associated with "{self.python_path}" aren\'t '
-                f'compatible with the "{virtualenv_name}" virtualenv'
-            )
 
         return result.provides_any_package
 
-- 
2.35.1

