From 61e40f477746888fefca3e89ce55e2f1448dd195 Mon Sep 17 00:00:00 2001
From: Amit Prakash Ambasta <amit.prakash.ambasta@gmail.com>
Date: Fri, 11 Mar 2022 16:14:56 +0530
Subject: [PATCH 26/26] bmo#1753182: Resolve fs symlinks

Signed-off-by: Amit Prakash Ambasta <amit.prakash.ambasta@gmail.com>
---
 build/moz.configure/init.configure | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/build/moz.configure/init.configure b/build/moz.configure/init.configure
index be079d0a43..bdf2784150 100644
--- a/build/moz.configure/init.configure
+++ b/build/moz.configure/init.configure
@@ -136,11 +136,12 @@ shell = help_shell | shell
 @checking("for Python 3", callback=lambda x: "%s (%s)" % (x.path, x.str_version))
 @imports("sys")
 @imports(_from="mach.site", _import="PythonVirtualenv")
+@imports(_from="os.path", _import="realpath")
 def virtualenv_python3():
     return namespace(
         # sys.executable is currently not updated for in-process activations. However,
         # sys.prefix is, so we can calculate the python executable's path from there.
-        path=normsep(PythonVirtualenv(sys.prefix).python_path),
+        path=normsep(PythonVirtualenv(realpath(sys.prefix)).python_path),
         str_version=".".join(str(i) for i in sys.version_info[0:3]),
     )
 
-- 
2.35.1

