From 6def7dbfe78e0a7970839830a83bae100eda61c7 Mon Sep 17 00:00:00 2001
From: Amit Prakash Ambasta <amit.prakash.ambasta@gmail.com>
Date: Mon, 14 Mar 2022 15:31:53 +0530
Subject: [PATCH] Move libmozgtk to os_libs for as_needed

Signed-off-by: Amit Prakash Ambasta <amit.prakash.ambasta@gmail.com>
---
 toolkit/library/moz.build | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/toolkit/library/moz.build b/toolkit/library/moz.build
index 0c02a1c760..3d28c1557e 100644
--- a/toolkit/library/moz.build
+++ b/toolkit/library/moz.build
@@ -182,9 +182,18 @@ if CONFIG["MOZ_WIDGET_TOOLKIT"] == "gtk":
     # get symbols from libwayland-client too.
     # In the case where Gtk+ doesn't have wayland support, libwayland-client is
     # not loaded, and libxul ends up using the mozwayland symbols.
-    USE_LIBS += [
+
+    if CONFIG["GCC_USE_GNU_LD"]:
+        no_as_needed = ["-Wl,--no-as-needed"]
+        as_needed = ["-Wl,--as-needed"]
+    else:
+        no_as_needed = []
+        as_needed = []
+    OS_LIBS += no_as_needed
+    OS_LIBS += [
         "mozgtk",
     ]
+    OS_LIBS += as_needed
     OS_LIBS += CONFIG["MOZ_GTK3_LIBS"]
 
 if CONFIG["MOZ_WAYLAND"]:
-- 
2.35.1

